frontend/src/components/FundsPullPreApproval/FundsPullPreApproval.tsx

```diff
@@ -0,0 +1,83 @@
+import { Approval } from "../../interfaces/approval";
+import React from "react";
+import { classNames } from "../../utils/class-names";
+import { Button, Col, Container, Row } from "reactstrap";
+import ApprovalDetails from "./ApprovalDetails";
+
+interface ApprovalProps {
+  approval: Approval;
+  onApproveClick: () => void;
+  onRejectClick: () => void;
+  onRevokeClick: () => void;
+  onAnyClickSetApproval: () => void;
+  displayApproveRejectButtons: boolean;
+  displayRevokeButton: boolean;
+}
+
+function FundsPullPreApproval({
+  approval,
+  onApproveClick,
+  onRejectClick,
+  onRevokeClick,
+  onAnyClickSetApproval,
+  displayApproveRejectButtons,
+  displayRevokeButton,
+}: ApprovalProps) {
+  const itemStyles = {
+    "list-group-item": true,
+  };
+
+  const onAnyClick = (method: () => void) => () => {
+    method();
+    onAnyClickSetApproval();
+  };
+
+  return (
+    <li className={classNames(itemStyles)} key={approval.funds_pull_pre_approval_id}>
+      <Container>
```

Comment:
Container inside `li`? you sure you need it?

Reply:
@vladkucherov - No, but I do need the Row

Reply:
It's OK to use row without a container